lib/smb2/client.rb

```diff
@@ -157,10 +160,42 @@ def negotiate
 
     @state = :negotiated
 
-    @sequence_number = 0
     # XXX do we need the Server GUID?
   end
 
+  # @param tree [String]
```

Comment:
Public methods should have descriptions in their docs.  `@return` is also missing.  Might want to add an `@see` to the Microsoft docs too.